Project3-Final

Pockyyyyyy opened the pull request:
# Project 3
Your Glitch project URL: Your-url-here

## Grading option
Choose **one** of the following by marking it with a ✅:

- [✅ ] I want detailed feedback on my GitHub PR
_this option is similar to how I have graded your previous projects_

- [ ] I do not require comments on GitHub, just a grade
_selecting this option, you will get a grading rubric sheet, but no comments on GitHub_

## Long(er) answers

> Describe something you struggled with during this project and how you overcame it.

I want to put in more designs, but I don't have more time, and I don't have a thorough understanding of label, so I want to make a better-looking CTA plate, but I don't have the time and energy. If I can, I wish I could spend more time on this assignment.

> Tell me what you're most proud of from project 3.

I learned how to use media, which is my happiest thing. I think the other one is that my design in title is the satisfactory part. Before doing this project, my HTML and CSS were rather messy, but now they have become much cleaner and many unnecessary DIVs have been removed.

> If you had another week or two to work on this project, what else would you do?

Make greater improvements in speakers, improve subtitle, improve register panel, add schedule panel, and redraw the text area.

> Share a new CSS property or technique you used on this project that you did not know or use before.

Li and ul.


## Short(er) answers

> Summarize the project requirements and evaluate if you have met them.

- ✅375 px for mobile, 1280 px for other devices
- ✅one-page site
- ✅event name
- ✅in-page navigation
- ✅event schedule
- ✅details for at least three talks
- ✅three speaker bios
- ✅call to action and link to buy tickets
- ✅header main footer h1 img a
- ✅two media queries
- ✅two css variables
- ✅sources

> Did you review the best practices checklists for [HTML](https://mica-web.github.io/learn/html/best-practices) and [CSS](https://mica-web.github.io/learn/css/best-practices) to check your own work on this project?

kind of.

----------------------------------------------------------------------------------------------------------------------------------------------

Alex Jaffe bio, speech, information: https://www.gdcvault.com/play/1025756/Cursed-Problems-in-Game
Stone Librande bio, speech, information: https://www.gdcvault.com/play/1022001/game
Jennifer Scheurle bio, speech, information: https://www.gdcvault.com/play/1024959/Good-Game-Design-is-like

Commit:
⚡️📚 Checkpoint
./event.html:22059420/217
./index.html:22059420/55

Commit:
👩‍👩‍👦🐟 Checkpoint
./index.html:22059420/340

Commit:
💶🏍 Checkpoint
./index.html:22059420/3938

Commit:
🙆🎐 Checkpoint
./index.html:22059420/1827

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,66 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="index.css">
+    <title>Document</title>
```
Remember to come back and update your `title` element; the conference name would make sense here!

angeliquejw replied:
Your `head` is missing a [proper `title` element](https://mica-web.github.io/learn/html/best-practices#include-a-proper-title-in-your-html-head). This is to add document information to browser windows and tabs and when a user bookmarks a site, among other things. It can include no other HTML or code, but it can include spaces, words or numbers.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,66 @@
+</head>
+
+<body>
+Dec 20th, 10.30 AM EST (Zoom)
```
You have the content all sorted here, but not the expected markup -- There should be more HTML providing meaning and structure to this content. Let me know if you want more specific feedback or other examples to look at?

angeliquejw reviewed (commented):
Hi, Pocky! The base of the HTML page is started, but it looks like there's still more work to be done adding HTML elements around your content.

Commit:
👨‍❤️‍💋‍👨💿 Checkpoint
./index.html:22059420/278
./index.css:22059420/167

Commit:
💺❄️ Checkpoint
./index.css:22059420/542
./index.html:22059420/85

Commit:
✨🌷 Checkpoint
./index.html:22059420/577

Commit:
🌅🗽 Checkpoint
./index.html:22059420/3059
./index.css:22059420/1129

Commit:
🐑⚓️ Checkpoint
./index.html:22059420/1027
./.glitch-assets:22059420/1705

Commit:
🏀🍐 Checkpoint
./index.html:22059420/2182
./index.css:22059420/553

Commit:
☄️🎢 Checkpoint
./index.html:22059420/3004
./index.css:22059420/549

Commit:
📚🛣 Checkpoint
./index.css:22059420/152
./index.html:22059420/2538

Commit:
🍛🚧 Checkpoint
./index.css:22059420/1968
./index.html:22059420/175

Commit:
📚🍨 Checkpoint
./.glitch-assets:22059420/763
./index.css:22059420/1858
./index.html:22059420/1137

Commit:
✈️🏪 Checkpoint
./index.css:22059420/1165

Commit:
👜🏮 Checkpoint
./index.css:22059420/1571
./index.html:22059420/672

Commit:
🐠🌝 Checkpoint
./index.html:22059420/688
./index.css:22059420/505

Commit:
🍴🔮 Checkpoint
./index.css:22059420/105

Commit:
🌦🐧 Checkpoint
./index.html:22059420/175
./index.css:22059420/275

Commit:
🙍🍥 Checkpoint
./index.css:22059420/564
./index.html:22059420/225

Commit:
🍛🚅 Checkpoint
./index.css:22059420/1063
./index.html:22059420/81

Commit:
🐀🍲 Checkpoint
./index.css:22059420/622
./index.html:22059420/709

Commit:
🐘🌒 Checkpoint
./index.css:22059420/98
./index.html:22059420/145

Commit:
⛳️👹 Checkpoint
./index.html:22059420/250
./index.css:22059420/5440
./.glitch-assets:22059420/911

Commit:
🐀🍽 Checkpoint
./index.html:22059420/313
./index.css:22059420/328

Commit:
👩‍👩‍👦🐊 Checkpoint
./index.css:22059420/781
./index.html:22059420/688

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+
+<body>
+    <main>
+        <div class="nav">
```
You could use a `nav` element here instead of `div`.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+    <main>
+        <div class="nav">
+            <ul class="topnav">
+                <li><a href="home">Home</a></li>
+                <li><a href="tickets">Tickets</a></li>
+                <li><a href="schedule">Schedule</a></li>
+                <li style="float:right"><a href="about">About</a></li>
```
While you have link elements in place here, the navigation itself is not functional. This [reading and demo](https://mica-web.github.io/learn/html/links#specific-elements) described how to create in-page links, as well as [this demo](https://glitch.com/~w12-demo) from week 12.

angeliquejw replied:
Your CSS and HTML should be separate; so all your CSS should be in your external `index.css`

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+                    class="big2">Games?</div>
+            </h1>
+            <div class="strong">
+                <h2><a href="">Workshop+Talk w/ Three Famous Game Designers</a></h2>
+                <h2><a href="">Dec 20th, 10.30 AM EST (Zoom)</a></h2>
+                <h2><a href="">These events are free and open to the public.
+                        No prior game design experience required.</a></h2>
```
1. It's not clear to me where these links would go to in the finished page?
2. The text on lines 28-29 should likely be in a `p` element vs a heading.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+                        No prior game design experience required.</a></h2>
+            </div>
+
+            <h4>Game Designers craft the core mechanics of a game.
```
1. You have skipped a heading level here -- there is no `h3` in your structure; see the HTML best practices re: [heading elements](https://mica-web.github.io/learn/html/best-practices#heading-tags-are-used-semantically-not-simply-for-presentation).
2. This text may better be served by a `p` element as it has multiple sentences in it and isn't really a heading/headline?

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+        </div>
+        <ul class="u-list-reset speaker-group">
+            <li class="speakers">
+                <div class="author-company">
+                    <img src="https://cdn.glitch.com/46be344e-bd86-488e-bf75-ab4ed5228fee%2Falex_jaffe-2c0a51ae.jpg?v=1608230332266"
+                        alt="Alex Jaffe" class="author-img">
```
This `alt` attribute (and those below) only repeat what can already be learned from the headings that follow -- what other meaningful information does this photo convey to a sighted user that should also get passed onto someone using a screen reader? See my notes on [writing good `alt` text](https://mica-web.github.io/learn/html/images#writing-good-alt-text).

angeliquejw replied:
Using `author-` prefixed class names here is confusing when everything else refers to these individuals as speakers; consider keeping your class names more consistent.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+            </h4>
+        </div>
+        <ul class="u-list-reset speaker-group">
+            <li class="speakers">
```
You have a lot of extra markup in this `li` that is not necessary to create this layout. Compare instead this example which has a lot fewer `div`s and other markup:
https://glitch.com/edit/#!/pocky-remix-p3-layout?path=index.html

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+                    <img src="https://cdn.glitch.com/46be344e-bd86-488e-bf75-ab4ed5228fee%2Falex_jaffe-2c0a51ae.jpg?v=1608230332266"
+                        alt="Alex Jaffe" class="author-img">
+                    <div class="speaker-title">
+                        <h4>Alex Jaffe</h4>
+                        <h5>A game Designer in R&D at Riot Games.</h5>
+                        <h4> <a href="https://www.gdcvault.com/play/1025756/Cursed-Problems-in-Game">Cursed Problems in
+                                Game Design</a></h4>
```
Again, you've skipped a heading level here; there is no `h3` in your document.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+                                Game Design</a></h4>
+                    </div>
+                </div>
+                <footer>
+                    <p> Game designers love solving problems, perfecting games until they
+                        sing. But sometimes you hit a design problem so thorny that you
+                        just can't find a solution, and maybe never will. This may be a
+                        "cursed problem": a problem with some inherent contradiction of
+                        goals. Consider the kingmaker problem. A designer wants to
+                        emphasize certain player skills, but also wants free-for-all PvP,
+                        which incentivizes politics in favor of those skills. Cursed problems
+                        like these haunt designers across game and genre. And somehow
+                        all designers have to deal with them.In this talk, Alex Jaffe (Riot
+                        Games) will present a theory of cursed problems. He'll show case
+                        studies of decades-old problems, and he'll explore the four design
+                        paradigms that have helped developers face these problems and
+                        survive.
+                    </p>
+                </footer>
```
If there is consistently only one item inside this `footer`, you can eliminate the `footer` element and just style the `p` element directly.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+                </footer>
+            </li>
+
+            <li class="speakers">
```
All of my comments about lines 46-73 above are true for the next two `li` elements as well.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,138 @@
+        </ul>
+        <footer class="register">
+            <form action="">
+                <label for="email">Enter You Email to Register!</label><br /><br />
+                <input type="text" id="email" name="email" value="" /><br /><br />
```
There is a separate `input` type just for email 👍🏻 
```suggestion
                <input type="email" id="email" name="email" value="" /><br /><br />
```

angeliquejw replied:
I know the demo of `form` code used `br` elements here, but you can and should control the layout of this form via CSS instead of `br ` elements.

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+    font-weight: 400;
+    line-height: 1.3;
+    margin: 32px;
+    font-family: Arial, Arial, Helvetica, sans-serif;
```
Your font stack repeats `Arial` twice:
```suggestion
    font-family: Arial, Helvetica, sans-serif;
```

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+}
+
+.nav {
+    box-sizing: border;
```
1. `border` is not a valid value for this property; you want `border-box`;
2. Instead of using this on individual elements, consider applying `box-sizing` to the entire stylesheet; see my [best practices list](https://mica-web.github.io/learn/css/best-practices#declare-box-sizing-in-all-your-stylesheets).

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+}
+
+.topnav a {
+    float: left;
```
I usually reserve floating for elements that need to have text wrapped around them; here, instead, the effect you want is to have side-by-side elements and `display:inline-block` will get you that. You would need to apply this to the `li` instead of the `a` element here.

Also, you have floated an element without a [clearfix](https://mica-web.github.io/learn/css/best-practices#wrap-all-floated-elements-in-another-element-with-the-clearfix-class-applied) applied to the parent.

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+    padding: 14px 16px;
+    text-decoration: none;
+    font-size: 17px;
+}
```
Consider adding a `transition` to the `a` element here to make your hover effect below smoother.

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+}
+
+.topnav {
+    list-style-type: none;
```
On lines 41-46 below, you have the `u-list-reset` class, but here you also repeat its styles. Instead of repeating those styles here, you could apply two classes to your `ul` in the HTML file, like so:
```html
<ul class="u-list-reset topnav">
```

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+}
+
+.title {
+    box-sizing: border;
```
See previous note about `box-sizing`

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+.title {
+    box-sizing: border;
+    padding: 10px;
+    margin-top: -30px;
+    margin-bottom: -10px;
```
You're relying on negative margins here to fix what should instead be addressed by undoing the default margins on the `h1` element. Also...you're adding 10px of padding to the top and bottom, then removing it with the margins. Even if this gets you the look you want, there are simpler, more straight forward ways to do so and you should work toward those.

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+    margin-right: 5px;
+}
+
+.big1 {
+    font-size: 50px;
+    background-color: #333;
+    width: 18%;
+    color: white;
+}
+
+.big2 {
+    font-size: 50px;
+    background-color: #333;
+    width: 15%;
+    color: white;
+}
```
If the only thing different about these elements is their width, consider using a single shared class and instead setting the width to be `width: max-content`. This would force the background color to only be as wide as the text inside it.

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+}
+
+.strong a {
+    font-family: Arial, Helvetica, sans-serif;
```
This only repeats the styles already set on line 7 above.
```suggestion
```

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+.author-img {
+    border-radius: 50%;
+    height: auto;
+    margin-right: auto;
+    margin-bottom: 12px;
+    margin-left: auto;
```
The `margin-left` and `margin-right` rules here aren't having any effect:
```suggestion
    margin-bottom: 12px;
```

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+}
+
+.author-company {
+    font-family: Arial, Helvetica, sans-serif;
```
This duplicates the existing rule on line 7.
```suggestion
```

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+}
+
+.speakers,
+.speakers fotter {
```
```suggestion
.speakers footer {
```

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+
+.speakers,
+.speakers fotter {
+    border-radius: 0 0 -6px 6px;
```
As you can see here in dev tools, this rule isn't working; negative numbers are not allowed in `border-radius`:
<img width="993" alt="Screen Shot 2020-12-19 at 4 33 51 PM" src="https://user-images.githubusercontent.com/2653980/102700069-168e3980-4218-11eb-8218-d98b283ee23f.png">

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+    color: white;
+    margin: 20px -12px 10px;
+    padding: 12px;
+    text-align: left;
```
This reiterates the  [default browser styles](https://browserdefaultstyles.com/) for `footer` and, therefore, can be removed:
```suggestion
```

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+}
+
+.speaker-title {
+    box-sizing: border;
```
See previous note about `box-sizing`

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+
+.speaker-title {
+    box-sizing: border;
+    margin-top: -140px;
+    margin-left: 140px;
```
See the Glitch example I shared previously for how to avoid using negative margins like this in your layout.

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+    box-sizing: border;
+    margin-top: -140px;
+    margin-left: 140px;
+    text-align: left;
```
This reiterates the  [default browser styles](https://browserdefaultstyles.com/) for `div` and, therefore, can be removed:
```suggestion
```

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+}
+
+.speakers {
+    background-size: 30%;
```
This rule could be added to the selector above?

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+    }
+
+    .speakers footer {
+        font-size: 12px;
```
This text size is too small for information you actually want conference attendees to read and benefit from. In general, body text on the web should be 16-18px.

angeliquejw commented on index.css:
```diff
@@ -0,0 +1,186 @@
+        margin: 20px -12px 10px;
+    }
+
+    .speaker-title {
```
Throughout your media queries, keep in mind:
1. My earlier comments about using margins to move things around for layout when there are other options, like `display: inline-block`.
2. Repeating browser defaults for spacing, text alignment, etc.
3. Dropping your text sizes below 15px for important content.
4. You only need to include styles in your media queries that are new or different; so, if the text color is white at all sizes, it does not need to be reiterated here; your earlier CSS will still control that. 👍🏻
5.

angeliquejw reviewed (approved):
Hi, Pocky, thanks for your work on Project 3. I've left some feedback about ways you can improve your code in the review below and talked through some of your design choices in this video: https://www.loom.com/share/86eb13a6221a4d3480de4cd2276c146a
